src/openrct2/world/Climate.cpp

```diff
@@ -205,13 +205,18 @@ void ClimateUpdate()
         ClimateUpdateLightning();
         ClimateUpdateThunder();
     }
-    else if (
-        gameState.weatherCurrent.weatherEffect == WeatherEffectType::Storm
-        || gameState.weatherCurrent.weatherEffect == WeatherEffectType::Blizzard)
+    else
     {
+        uint32_t thunderChance;
+        if (gameState.weatherCurrent.weatherEffect == WeatherEffectType::Storm)
+            thunderChance = 0x1B4;
+        else if (gameState.weatherCurrent.weatherEffect == WeatherEffectType::Blizzard)
+            thunderChance = 0x6D;
+        else
+            return;
         // Create new thunder and lightning. Their amount is scaled inversely proportional
         // to the game speed, otherwise they become annoying at very high speeds
-        if (uint32_t randomNumber = UtilRand(); (randomNumber & 0xFFFF) <= (0x1B4u >> (gGameSpeed - 1)))
+        if (uint32_t randomNumber = UtilRand(); (randomNumber & 0xFFFF) <= thunderChance >> (gGameSpeed - 1))
```

Comment:
```suggestion
        if (uint32_t randomNumber = UtilRand(); (randomNumber & 0xFFFF) <= (thunderChance >> (gGameSpeed - 1)))
```
While this should work as `>>` [take precedence](https://en.cppreference.com/w/cpp/language/operator_precedence.html) over `<=`, I think the parentheses make it easier to understand for humans.